File: aguaegas-api/aguaegas/src/main/java/com/project/aguaegas/domain/model/Empresario.java
package com.project.aguaegas.domain.model;

import com.project.aguaegas.domain.exception.TipoUsuario;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.usertype.UserType;


@Entity
@Data
public class Empresario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    private String email;
    private String senha;

    @Enumerated(EnumType.STRING)
    private TipoUsuario tipoUsuario;

    public String getPassword() {
        return senha;
    }
}

File: aguaegas/lib/api/models/empresario.dart
class Empresario {
  final int id;
  final String nome;
  final String email;
  final String senha;

  Empresario({
    required this.id,
    required this.nome,
    required this.email,
    required this.senha,
  });

  factory Empresario.fromJson(Map<String, dynamic> json) {
    return Empresario(
      id: json['id'],
      nome: json['nome'],
      email: json['email'],
      senha: json['senha'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nome': nome,
      'email': email,
      'senha': senha,
    };
  }
}
